### PR TITLE
Adds docs for AmbientAware

### PR DESCRIPTION
**Pull request by garanj**
#### WHAT

Adds docs for `AmbientAware`

#### WHY

To help developers understand how to use `AmbientAware`

#### HOW

N/A

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: docs/compose-layout.md [hunk 1]
```diff
@@ -85,3 +113,6 @@ dependencies {
     implementation "com.google.android.horologist:horologist-compose-layout:<version>"
 }
 ```
+
+
+[always-on]: https://developer.android.com/training/wearables/views/always-on
```
Comment (yschimke) [inline docs/compose-layout.md:118]: Worth also linking to Exercise instructions? https://developer.android.com/training/wearables/health-services/active-data#work-with-data

File: docs/compose-layout.md [hunk 2]
```diff
@@ -74,6 +74,34 @@ Box(
...
 
+## AmbientAware composable
+
+`AmbientAware` allows your UI to react to ambient mode changes. For more information on how Ambient
+mode and Always-on work on Wear OS, see the [developer guidance][always-on].
+
+You should place this composable high up in your design, as it alters the behavior of the Activity.
+
+```kotlin
+WearApp {
```
Comment (yschimke) [inline docs/compose-layout.md:85]:
```suggestion
@Composable
fun WearApp() {
```

File: docs/compose-layout.md (hunk 1 above)
Comment (yschimke) [reply docs/compose-layout.md:118]: Also to health-samples?

**Review by yschimke — APPROVED** [review]

File: docs/compose-layout.md (hunk 1 above)
Comment (garanj) [reply docs/compose-layout.md:124]: Done

**Pull request merged**